### DISCUSSION TITLE
Aggregate Interval & Aggregate Feature in readProcessed

### DISCUSSION THREAD

**Discussion by gilsalvans** [duduyoyo/WebSocket4OPC#14] [General]
Hello, 
I have been testing around with different read methods, and I have a couple of questions regarding the **readProcessed** query method:

1. The "Aggregate Interval" is limited to maximum 120000 Milliseconds (2 minutes). Would there be any possiblity to extend this to larger aggregation timespans, such as 10 minutes or even an hour?

2. The "Aggregate Feature" parameter says that it must be a value between 1 and 24. By playing around, I have figured out that a 3 gives an average, an 8 a minimum or a 10 a maximum. However, is there any documentation with all options and what does each value (between 1 and 24) exactly gives out as aggregation output?

Thank you in advance
Best
Gil

**Comment by duduyoyo** [comment]
My implementation did impose a limit of 2 minutes for interval. When you see this, what is quality status in return? You need check with HDA spec to find out what that status means and take action accordingly. Also, you can use 3rd party client tool like Marticon HDA Explorer to verify.

Chapter 5.3.3 of HDA spec gives you a list of all 24 aggregate methods. Please check there.

If you plan to apply your application in a production environment it would be the best practice purchasing a license to remove restriction on trial edition. You can contact at websocket4opc@gmail.com. Cheers!

**Comment by duduyoyo** [comment]
![image](https://github.com/user-attachments/assets/0abbbe60-5d4b-4f33-bb59-bb2c18153a4a)
